src/Heap.sol

```diff
@@ -36,209 +40,202 @@ library BasicHeap {
     /// @param _heap The heap to modify.
     /// @param _id The address of the account to insert.
     /// @param _value The value of the account to insert.
-    function insert(
-        Heap storage _heap,
-        address _id,
-        uint256 _value
-    ) internal {
-        if (_value == 0) revert WrongValue();
+    function insert(Heap storage _heap, address _id, uint256 _value) internal {
         if (_id == address(0)) revert AddressIsZero();
-        if (_heap.ranks[_id] != 0) revert AccountAlreadyInserted();
 
-        // Put the account at the end of the heap.
-        _heap.accounts.push(Account(_id, _value));
-        uint256 size = getSize(_heap);
-        _heap.ranks[_id] = size;
+        uint256 accountsLength = _heap.accounts.length;
+        if (containsAccount(_heap, _heap.indexOf[_id], accountsLength, _id)) revert AccountAlreadyInserted();
+
+        _heap.accounts.push();
 
-        // Restore the invariant.
-        shiftUp(_heap, size);
+        shiftUp(_heap, Account(_id, _value), accountsLength);
     }
 
     /// @notice Decreases the amount of an account in the `_heap`.
     /// @param _heap The heap to modify.
     /// @param _id The address of the account to decrease the amount.
     /// @param _newValue The new value of the account.
-    function decrease(
-        Heap storage _heap,
-        address _id,
-        uint256 _newValue
-    ) internal {
-        uint256 rank = _heap.ranks[_id];
-        if (rank == 0) revert AccountDoesNotExist();
-        uint256 oldValue = getAccount(_heap, rank).value;
-        if (_newValue >= oldValue || _newValue == 0) revert WrongValue();
-
-        setAccountValue(_heap, rank, _newValue);
-        shiftDown(_heap, rank);
+    function decrease(Heap storage _heap, address _id, uint256 _newValue) internal {
+        uint256 index = _heap.indexOf[_id];
+
+        if ((!containsAccount(_heap, index, _heap.accounts.length, _id))) revert AccountDoesNotExist();
```

Comment:
```suggestion
        if (!containsAccount(_heap, index, _heap.accounts.length, _id)) revert AccountDoesNotExist();
```

There's still the unnecessary parenthesis no?

Reply:
you're right mb